packages/suite/src/actions/suite/metadataActions.ts

```diff
@@ -684,9 +724,31 @@ export const addMetadata = (payload: MetadataAddPayload) => (dispatch: Dispatch)
 export const init =
     (force = false) =>
     async (dispatch: Dispatch, getState: GetState) => {
+        // 1. connect to provider
+        if (!getState().metadata.provider) {
+            if (!getState().metadata.initiating) {
+                dispatch({ type: METADATA.SET_INITIATING, payload: true });
+            }
+
+            const providerResult = await dispatch(initProvider());
+            if (!providerResult) {
+                dispatch({ type: METADATA.SET_INITIATING, payload: false });
+                dispatch({ type: METADATA.SET_EDITING, payload: undefined });
+
+                return false;
+            }
+        }
+
+        // 2. get the metadata minimum version to determine a migration is needed
+        const metadataMimimumVersion = await dispatch(getMetadataFilesMinimumVersion());
+
+        if (!metadataMimimumVersion) {
+            return false;
```

Comment:
do I read it correct that in case I don't need migration, lines below this return are not executed? maybe I will find why later